### PR TITLE
Enable Linux release from ADO.

### PR DESCRIPTION
**Pull request by goagain**
This PR contains 2 changes.
- Enable building Linux packages from ADO pipelines again. It will generate both `amd64` and `arm64` packages.
- The ADO pipeline is staged not. We don't need to wait for the publish stage. After the `Sign` stage completed, We can download the artifacts.

### CODE REVIEW

**Comment by goagain** [comment]
@AzureAD/security-experience The pipeline has been validated. Please have a review.

File: bin/trigger_azure_pipelines.py [hunk 1]
```diff
@@ -10,12 +10,13 @@
 
 from azure.devops.connection import Connection
 from azure.devops.v6_0.pipelines.pipelines_client import PipelinesClient
-from azure.devops.v6_0.pipelines.models import Run
+from azure.devops.v6_0.build.build_client import BuildClient
+from azure.devops.v6_0.build.models import TimelineRecord
 from msrest.authentication import BasicAuthentication
 from requests import Response
 
 # https://learn.microsoft.com/en-us/rest/api/azure/devops/pipelines/runs/get?view=azure-devops-rest-6.0#runresult
```
Comment (mvanchaa) [inline bin/trigger_azure_pipelines.py:18]: @goagain can you update the comment here with the right reference link?

File: bin/trigger_azure_pipelines.py [hunk 2]
```diff
@@ -10,12 +10,13 @@
...
 from azure.devops.v6_0.pipelines.pipelines_client import PipelinesClient
-from azure.devops.v6_0.pipelines.models import Run
+from azure.devops.v6_0.build.build_client import BuildClient
+from azure.devops.v6_0.build.models import TimelineRecord
 from msrest.authentication import BasicAuthentication
 from requests import Response
 
 # https://learn.microsoft.com/en-us/rest/api/azure/devops/pipelines/runs/get?view=azure-devops-rest-6.0#runresult
-FAILED_STATUSES: set[str] = {"canceled", "failed", "unknown"}
+FAILED_STATUSES: set[str] = {"canceled", "failed", "skipped", "unknown"}
```
Comment (mvanchaa) [inline bin/trigger_azure_pipelines.py:19]: There is another status called `abandoned`. And `unknown` status doesn't exist. https://learn.microsoft.com/en-us/rest/api/azure/devops/build/timeline/get?view=azure-devops-rest-6.0#taskresult. I think we should handle that scenario as well.

**Review by mvanchaa — CHANGES_REQUESTED** [review]

**Review by mvanchaa — DISMISSED** [review]

File: bin/trigger_azure_pipelines.py [hunk 3]
```diff
@@ -27,31 +30,35 @@ def ado_connection(organization: str, ado_pat: str) -> Connection:
...
-    while run.state not in COMPLETED_STATUSES:
-        time.sleep(polling_interval_seconds)
-        run = pipeline_client.get_run(project, pipeline_id, run_id)
-    return run
+    # Wait until the stage have a complete status.
+    # https://learn.microsoft.com/en-us/rest/api/azure/devops/build/timeline/get?view=azure-devops-rest-6.0
+    while True:
+        timeline = build_client.get_build_timeline(project, run_id)
+        records = timeline.records
+        record = next(filter(lambda record: record.identifier == stage_id, records), None)
```
Comment (reillysiemens) [inline bin/trigger_azure_pipelines.py:47]:
```suggestion
        record = next((r for r in records if r.identifier == stage_id), None)
```

File: bin/trigger_azure_pipelines.py [hunk 4]
```diff
@@ -27,31 +30,35 @@ def ado_connection(organization: str, ado_pat: str) -> Connection:
...
-    # https://learn.microsoft.com/en-us/rest/api/azure/devops/pipelines/runs/get?view=azure-devops-rest-6.0#runstate
-    while run.state not in COMPLETED_STATUSES:
-        time.sleep(polling_interval_seconds)
-        run = pipeline_client.get_run(project, pipeline_id, run_id)
-    return run
+    # Wait until the stage have a complete status.
+    # https://learn.microsoft.com/en-us/rest/api/azure/devops/build/timeline/get?view=azure-devops-rest-6.0
+    while True:
+        timeline = build_client.get_build_timeline(project, run_id)
+        records = timeline.records
```
Comment (reillysiemens) [inline bin/trigger_azure_pipelines.py:46]: If you take my suggestion below you don't need this line.

**Review by reillysiemens — DISMISSED** [review]

File: bin/trigger_azure_pipelines.py (hunk 4 above)
Comment (goagain) [inline bin/trigger_azure_pipelines.py:46]:
```suggestion
```

**Review by mvanchaa — APPROVED** [review]

**Review by reillysiemens — APPROVED** [review]

**Pull request merged**